src/task.test.js

```diff
@@ -70,4 +78,74 @@ describe('deleteTask', () => {
     expect(tasks[0]).toEqual(task1);
     expect(tasks[1]).toEqual(task3);
   });
+});
+
+describe('updateTaskValue', () => {
+  beforeEach(() => {
+    localStorage.clear(); // Clear localStorage before each test
+  });
+
+  test('Should update the task value', () => {
+    // Create a test task object
+    const task = {
+      description: 'Task 1',
+      completed: false,
+      index: 1,
+    };
+
+    // Create a new label element to use as the label parameter
+    const label = document.createElement('input');
+    label.value = 'New description';
+
+    // Call updateTaskValue() with the test task and label
+    updateTaskValue(task, label);
+
+    // Check if the task description has been updated correctly
+    expect(task.description).toEqual('New description');
+  });
+});
+
+describe('AddCheckBox', () => {
+  beforeEach(() => {
+    localStorage.clear(); // Clear localStorage before each test
+  });
+
+  test('it should update the task completed status when checkbox is checked/unchecked', () => {
+    const checkbox = document.createElement('input');
+    const task = { description: 'Task 1', completed: false };
+    const listItem = document.createElement('li');
+
+    AddCheckBox(checkbox, task, listItem);
+    checkbox.checked = true;
+    // trigger the event listener for the 'change' event on the checkbox element
+    checkbox.dispatchEvent(new Event('change'));
+
+    expect(task.completed).toBe(true);
+  });
+});
+
+describe('ClearCompletedTasks', () => {
+  beforeEach(() => {
+    localStorage.clear(); // Clear localStorage before each test
+  });
+
+  test('should remove completed tasks from the tasks array and localStorage', () => {
+    // Reset the tasks array to its initial state before each test
+    const task01 = { description: 'Task 01', completed: true, index: 0 };
+    const task02 = { description: 'Task 02', completed: false, index: 1 };
+    const task03 = { description: 'Task 03', completed: true, index: 2 };
+    localStorage.setItem('tasksObj', JSON.stringify([task01, task02, task03]));
+
+    /* eslint-disable global-require */
+    const { ClearCompletedTasks } = require('./completion.js');
+    /* eslint-enable global-require */
+    const tasksToCheck = JSON.parse(localStorage.getItem('tasksObj'));
+
+    ClearCompletedTasks(tasksToCheck);
+
+    const updatedTasks = JSON.parse(localStorage.getItem('tasksToCheck'));
+
+    expect(updatedTasks).toHaveLength(1);
+    expect(updatedTasks[0].completed).toEqual(false);
+  });
```

Comment:
-   [ ] Please consider testing  the mock DOM manipulation in the case of the function `ClearCompletedTasks` as well following the same example above 👍🏼